README.md
convert-units
=============

[![Build Status](https://travis-ci.org/ben-ng/convert-units.png)](https://travis-ci.org/ben-ng/convert-units) [![Downloads](https://img.shields.io/npm/dm/convert-units.svg)](https://www.npmjs.com/package/convert-units)

A handy utility for converting between quantities in different units.

Installation
-----

```
npm install convert-units --save
```

Usage
-----

`convert-units` has a simple chained API that is easy to read.

Here's how you move between the metric units for volume:

```js
var convert = require('convert-units')

convert(1).from('l').to('ml')
// 1000
```

Jump from imperial to metric units the same way:

```js
convert(1).from('lb').to('kg')
// 0.4536... (tested to 4 significant figures)
```

Just be careful not to ask for an impossible conversion:

```js
convert(1).from('oz').to('fl-oz')
// throws -- you can't go from mass to volume!
```

You can ask `convert-units` to select the best unit for you. You can also optionally explicitly exclude orders of magnitude or specify a cut off number for selecting the best representation.
```js
convert(12000).from('mm').toBest()
// 12 Meters (the smallest unit with a value above 1)

convert(12000).from('mm').toBest({ exclude: ['m'] })
// 1200 Centimeters (the smallest unit excluding meters)

convert(900).from('mm').toBest({ cutOffNumber: 10 });
// 90 Centimeters (the smallest unit with a value equal to or above 10)

convert(1000).from('mm').toBest({ cutOffNumber: 10 })
// 100 Centimeters (the smallest unit with a value equal to or above 10)
```

You can get a list of the measurement types supported with `.measures`

```js
convert().measures()
// [ 'length', 'mass', 'volume' ]
```

If you ever want to know the possible conversions for a unit, just use `.possibilities`

```js
convert().from('l').possibilities()
// [ 'ml', 'l', 'tsp', 'Tbs', 'fl-oz', 'cup', 'pnt', 'qt', 'gal' ]

convert().from('kg').possibilities()
// [ 'mcg', 'mg', 'g', 'kg', 'oz', 'lb' ]
```

You can also get the possible conversions for a measure:
```js
convert().possibilities('mass')
// [ 'mcg', 'mg', 'g', 'kg', 'oz', 'lb', 'mt', 't' ]
```

You can also get the all the available units:
```js
convert().possibilities()
// [ 'mm', 'cm', 'm', 'in', 'ft-us', 'us-ft', 'ft', 'mi', 'mcg', 'mg', 'g', 'kg', 'oz', 'lb', 'mt', 't', 'ml', 'l', 'tsp', 'Tbs', 'fl-oz', 'cup', 'pnt', 'qt', 'gal', 'ea', 'dz' ];
```

To get a detailed description of a unit, use `describe`

```js
convert().describe('kg')
/*
  {
    abbr: 'kg'
  , measure: 'mass'
  , system: 'metric'
  , singular: 'Kilogram'
  , plural: 'Kilograms'
  }
*/
```

To get detailed descriptions of all units, use `list`.

```js
convert().list()
/*
  [{
    abbr: 'kg'
  , measure: 'mass'
  , system: 'metric'
  , singular: 'Kilogram'
  , plural: 'Kilograms'
  }, ...]
*/
```

You can also get detailed descriptions of all units for a measure:

```js
convert().list('mass')
/*
  [{
    abbr: 'kg'
  , measure: 'mass'
  , system: 'metric'
  , singular: 'Kilogram'
  , plural: 'Kilograms'
  }, ...]
*/
```

Supported Units
---------------
### Length
* mm
* cm
* m
* in
* ft-us (us-ft)
* ft
* fathom
* mi
* nMi
* in-us (us-in)

### Area
* mm2
* cm2
* m2
* ha
* km2
* in2
* ft2
* ac
* mi2

### Mass
* mcg
* mg
* g
* kg
* oz
* lb
* mt
* t

### Volume
* mm3
* cm3
* ml
* l
* kl
* m3
* km3
* tsp
* Tbs
* in3
* fl-oz
* cup
* pnt
* qt
* gal
* ft3
* yd3

### Volume Flow Rate
* mm3/s
* cm3/s
* ml/s
* cl/s
* dl/s
* l/s
* l/min
* l/h
* kl/s
* kl/min
* kl/h
* m3/s
* m3/min
* m3/h
* km3/s
* tsp/s
* Tbs/s
* in3/s
* in3/min
* in3/h
* fl-oz/s
* fl-oz/min
* fl-oz/h
* cup/s
* pnt/s
* pnt/min
* pnt/h
* qt/s
* gal/s
* gal/min
* gal/h
* ft3/s
* ft3/min
* ft3/h
* yd3/s
* yd3/min
* yd3/h'

### Temperature
* C
* F
* K
* R

### Time
* ns
* mu
* ms
* s
* min
* h
* d
* week
* month
* year

### Frequency
* Hz
* mHz
* kHz
* MHz
* GHz
* THz
* rpm
* deg/s
* rad/s

### Speed
* m/s
* km/h
* m/h
* knot
* ft/s

### Pace
* s/m
* min/km
* s/ft
* min/km

### Pressure
* Pa
* hPa
* kPa
* MPa
* bar
* torr
* psi
* ksi

### Digital
* b
* Kb
* Mb
* Gb
* Tb
* B
* KB
* MB
* GB
* TB

### Illuminance
* lx
* ft-cd

### Parts-Per
* ppm
* ppb
* ppt
* ppq

### Voltage
* V
* mV
* kV

### Current
* A
* mA
* kA

### Power
* W
* mW
* kW
* MW
* GW

### Apparent Power
* VA
* mVA
* kVA
* MVA
* GVA

### Reactive Power
* VAR
* mVAR
* kVAR
* MVAR
* GVAR

### Energy
* Wh
* mWh
* kWh
* MWh
* GWh
* J
* kJ

### Reactive Energy
* VARh
* mVARh
* kVARh
* MVARh
* GVARh

### Angle
* deg
* rad
* grad
* arcmin
* arcsec

### Charge
* c
* mC
* μC
* nC
* pC

### Force
* N
* kN
* lbf

### Acceleration
* g (g-force)
* m/s2


### Want More?

Adding new measurement sets is easy. Take a look at [`lib/definitions`](https://github.com/ben-ng/convert-units/tree/master/lib/definitions) to see how it's done.

License
-------
Copyright (c) 2013-2017 Ben Ng and Contributors, http://benng.me

Permission is hereby granted, free of charge, to any person
obtaining a copy of this software and associated documentation
files (the "Software"), to deal in the Software without
restriction, including without limitation the rights to use,
copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the
Software is furnished to do so, subject to the following
conditions:

The above copyright notice and this permission notice shall be
included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
OTHER DEALINGS IN THE SOFTWARE.


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## v2.4.0
### Added
- This CHANGELOG
- Support for Us survey inches (us-in)




test/lengths.js
var convert = require('../lib')
  , assert = require('assert')
  , tests = {}
  , ACCURACY = 1/1000
  , percentError = require('../lib/percentError');


tests['ft to ft'] = function () {
  assert.strictEqual( convert(1).from('ft').to('ft') , 1);
};

tests['ft to ft-us'] = function () {
  assert.strictEqual( convert(1).from('ft').to('ft-us') , 0.999998000004);
};

tests['ft-us to ft'] = function () {
  assert.strictEqual( convert(1).from('ft-us').to('ft') , 1.000002);
};

tests['in to in'] = function () {
  assert.strictEqual( convert(7).from('in').to('in') , 7);
};

tests['ft to in'] = function () {
  assert.strictEqual( convert(1).from('ft').to('in') , 12);
};

tests['in to ft'] = function () {
  assert.strictEqual( convert(1).from('in').to('ft') , 1/12);
};

tests['ft to mi'] = function () {
  assert.strictEqual( convert(1).from('ft').to('mi') , 1/5280);
};

tests['mi to ft'] = function () {
  assert.strictEqual( convert(1).from('mi').to('ft') , 5280);
};

tests['nMi to mi'] = function () {
  assert.strictEqual( convert(1).from('nMi').to('mi') , 1.150780303030303);
};

tests['m to m'] = function () {
  assert.strictEqual( convert(1).from('m').to('m') , 1);
};

tests['m to cm'] = function () {
  assert.strictEqual( convert(1).from('m').to('cm') , 100);
};

tests['cm to m'] = function () {
  assert.strictEqual( convert(1).from('cm').to('m') , 1/100);
};

tests['m to mm'] = function () {
  assert.strictEqual( convert(1).from('m').to('mm') , 1000);
};

tests['km to m'] = function () {
  assert.strictEqual( convert(1).from('km').to('m'), 1000);
};

tests['us-in to in'] = function () {
  assert.strictEqual( convert(1).from('us-in').to('in'), 1.000002);
};

tests['in to us-in'] = function () {
  assert.strictEqual( convert(1).from('in').to('us-in'), 0.999998);
};

// When converting between systems, expect < 0.1% error
tests['m to ft'] = function () {
  var expected = 3.28084
    , actual = convert(1).from('m').to('ft');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

// When converting between systems, expect < 0.1% error
tests['m to ft-us'] = function () {
  var expected = 3.28084
    , actual = convert(1).from('m').to('ft-us');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

// When converting between systems, expect < 0.1% error
tests['m to us-ft'] = function () {
  var expected = 3.28084
    , actual = convert(1).from('m').to('us-ft');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['mm to ft'] = function () {
  var expected = 0.00328084
    , actual = convert(1).from('mm').to('ft');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['nMi to m'] = function () {
  var expected = 1852
    , actual = convert(1).from('nMi').to('m');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['km to nMi'] = function () {
  var expected = 0.539956803
    , actual = convert(1).from('km').to('nMi');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['fathm to m'] = function () {
  var expected = 1.8288
    , actual = convert(1).from('fathom').to('m');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['us-in to in'] = function () {
  var expected = 1.000002
    , actual = convert(1).from('us-in').to('in');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['in to us-in'] = function () {
  var expected = 0.999998
    , actual = convert(1).from('in').to('us-in');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['ft to us-in'] = function () {
  var expected = 12.000024
    , actual = convert(1).from('ft').to('us-in');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

tests['us-in toft'] = function () {
  var expected = 0.08333316667
    , actual = convert(1).from('in').to('us-ft');
  assert.ok( percentError(expected, actual) < ACCURACY
    , 'Expected: ' + expected +', Actual: ' + actual);
};

module.exports = tests;


test/possibilities.js
var convert = require('../lib')
  , assert = require('assert-diff')
  , tests = {};

assert.options.strict = true;

tests['l possibilities'] = function () {
  var actual = convert().from('l').possibilities()
    , expected = [ 'mm3', 'cm3', 'ml', 'cl', 'dl', 'l', 'kl', 'm3', 'km3', 'krm', 'tsk', 'msk', 'kkp', 'glas', 'kanna', 'tsp', 'Tbs', 'in3', 'fl-oz', 'cup', 'pnt', 'qt', 'gal', 'ft3', 'yd3' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['kg possibilities'] = function () {
  var actual = convert().from('kg').possibilities()
    , expected = [ 'mcg', 'mg', 'g', 'kg', 'mt', 'oz', 'lb', 't' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['m possibilities'] = function () {
  var actual = convert().from('m').possibilities()
    , expected = [ 'mm', 'cm', 'm', 'km', 'in', 'yd', 'ft-us', 'us-ft', 'ft', 'fathom', 'mi', 'nMi', 'us-in', 'in-us' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['each possibilities'] = function () {
  var actual = convert().possibilities('each')
    , expected = [ 'ea', 'dz' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['mass possibilities'] = function () {
  var actual = convert().possibilities('mass')
    , expected = [ 'mcg', 'mg', 'g', 'kg', 'mt', 'oz', 'lb', 't' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['volume possibilities'] = function () {
  var actual = convert().possibilities('volume')
    , expected = [ 'mm3', 'cm3', 'ml', 'cl', 'dl', 'l', 'kl', 'm3', 'km3', 'krm', 'tsk', 'msk', 'kkp', 'glas', 'kanna', 'tsp', 'Tbs', 'in3', 'fl-oz', 'cup', 'pnt', 'qt', 'gal', 'ft3', 'yd3' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['volume flow rate possibilities'] = function () {
  var actual = convert().possibilities('volumeFlowRate')
    , expected = [ 'mm3/s', 'cm3/s', 'ml/s', 'cl/s', 'dl/s', 'l/s', 'l/min', 'l/h', 'kl/s', 'kl/min', 'kl/h', 'm3/s', 'm3/min', 'm3/h', 'km3/s', 'tsp/s', 'Tbs/s', 'in3/s', 'in3/min', 'in3/h', 'fl-oz/s', 'fl-oz/min', 'fl-oz/h', 'cup/s', 'pnt/s', 'pnt/min', 'pnt/h', 'qt/s', 'gal/s', 'gal/min', 'gal/h', 'ft3/s', 'ft3/min', 'ft3/h', 'yd3/s', 'yd3/min', 'yd3/h' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['length possibilities'] = function () {
  var actual = convert().possibilities('length')
    , expected = [ 'mm', 'cm', 'm', 'km', 'in', 'yd', 'ft-us', 'us-ft', 'ft', 'fathom', 'mi', 'nMi', 'us-in', 'in-us' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['temperature possibilities'] = function () {
  var actual = convert().possibilities('temperature')
    , expected = ['C', 'K', 'F', 'R'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['time possibilities'] = function () {
  var actual = convert().possibilities('time')
    , expected = ['ns', 'mu', 'ms', 's', 'min', 'h', 'd', 'week', 'month', 'year'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['digital possibilities'] = function() {
  var actual = convert().possibilities('digital')
    , expected = [ 'b', 'Kb', 'Mb', 'Gb', 'Tb', 'B', 'KB', 'MB', 'GB', 'TB' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['partsPer possibilities'] = function() {
  var actual = convert().possibilities('partsPer')
    , expected = [ 'ppm', 'ppb', 'ppt', 'ppq' ];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['pressure possibilities'] = function() {
  var actual = convert().possibilities('pressure')
    , expected = [ 'Pa', 'kPa', 'MPa', 'hPa', 'bar', 'torr', 'psi', 'ksi'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['speed possibilities'] = function() {
  var actual = convert().possibilities('speed')
    , expected = [ 'm/s', 'km/h', 'm/h', 'knot', 'ft/s'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['pace possibilities'] = function() {
  var actual = convert().possibilities('pace')
    , expected = [ 's/m', 'min/km', 'min/mi', 's/ft'];
  assert.deepEqual(actual.sort(), expected.sort())
};

tests['current possibilities'] = function() {
  var actual = convert().possibilities('current')
    , expected = [ 'A', 'mA', 'kA'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['voltage possibilities'] = function() {
  var actual = convert().possibilities('voltage')
    , expected = [ 'V', 'mV', 'kV'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['power possibilities'] = function() {
  var actual = convert().possibilities('power')
    , expected = [ 'W', 'mW', 'kW', 'MW', 'GW'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['reactive power possibilities'] = function() {
  var actual = convert().possibilities('reactivePower')
    , expected = [ 'VAR', 'mVAR', 'kVAR', 'MVAR', 'GVAR'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['apparent power possibilities'] = function() {
  var actual = convert().possibilities('apparentPower')
    , expected = [ 'VA', 'mVA', 'kVA', 'MVA', 'GVA'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['energy possibilities'] = function() {
  var actual = convert().possibilities('energy')
    , expected = [ 'Wh', 'mWh', 'kWh', 'MWh', 'GWh', 'J', 'kJ'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['reactive energy possibilities'] = function() {
  var actual = convert().possibilities('reactiveEnergy')
    , expected = [ 'VARh', 'mVARh', 'kVARh', 'MVARh', 'GVARh'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['reactive energy possibilities'] = function() {
  var actual = convert().possibilities('frequency')
    , expected = [ 'Hz', 'mHz', 'kHz', 'MHz', 'GHz', 'THz', 'rpm', 'deg/s', 'rad/s'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['illuminance possibilities'] = function() {
  var actual = convert().possibilities('illuminance')
    , expected = [ 'lx', 'ft-cd'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['angle possibilities'] = function() {
  var actual = convert().possibilities('angle')
    , expected = [ 'rad', 'deg', 'grad', 'arcmin', 'arcsec'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['charge possibilities'] = function() {
  var actual = convert().possibilities('charge')
    , expected = [ 'c', 'mC', 'μC', 'nC', 'pC'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['force possibilities'] = function() {
  var actual = convert().possibilities('force')
    , expected = [ 'N', 'kN', 'lbf'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['acceleration possibilities'] = function() {
  var actual = convert().possibilities('acceleration')
    , expected = [ 'g-force', 'm/s2'];
  assert.deepEqual(actual.sort(), expected.sort());
};

tests['all possibilities'] = function () {
  var actual = convert().possibilities()
    // Please keep these sorted for maintainability
    , expected = [
        'A'
      , 'arcmin'
      , 'arcsec'
      , 'B'
      , 'C'
      , 'F'
      , 'R'
      , 'GB'
      , 'Gb'
      , 'K'
      , 'KB'
      , 'Kb'
      , 'MB'
      , 'MPa'
      , 'Mb'
      , 'N'
      , 'Pa'
      , 'TB'
      , 'Tb'
      , 'ac'
      , 'b'
      , 'bar'
      , 'c'
      , 'cl'
      , 'cl/s'
      , 'cm'
      , 'cm2'
      , 'cm3'
      , 'cm3/s'
      , 'cup'
      , 'cup/s'
      , 'd'
      , 'deg'
      , 'deg/s'
      , 'dl'
      , 'dl/s'
      , 'ea'
      , 'dz'
      , 'fl-oz'
      , 'fl-oz/h'
      , 'fl-oz/min'
      , 'fl-oz/s'
      , 'ft-us'
      , 'ft'
      , 'fathom'
      , 'ft-cd'
      , 'ft/s'
      , 'ft2'
      , 'ft3'
      , 'ft3/h'
      , 'ft3/min'
      , 'ft3/s'
      , 'g'
      , 'g-force'
      , 'gal'
      , 'gal/h'
      , 'gal/min'
      , 'gal/s'
      , 'glas'
      , 'grad'
      , 'GHz'
      , 'GVA'
      , 'GVAR'
      , 'GVARh'
      , 'GW'
      , 'GWh'
      , 'h'
      , 'hPa'
      , 'ha'
      , 'Hz'
      , 'in'
      , 'in2'
      , 'in3'
      , 'in3/h'
      , 'in3/min'
      , 'in3/s'
      , 'J'
      , 'kA'
      , 'kPa'
      , 'kanna'
      , 'kg'
      , 'kkp'
      , 'kJ'
      , 'kN'
      , 'kl'
      , 'kl/h'
      , 'kl/min'
      , 'kl/s'
      , 'km'
      , 'km/h'
      , 'km2'
      , 'km3'
      , 'km3/s'
      , 'knot'
      , 'krm'
      , 'ksi'
      , 'kHz'
      , 'kV'
      , 'kVA'
      , 'kVAR'
      , 'kVARh'
      , 'kW'
      , 'kWh'
      , 'l'
      , 'l/h'
      , 'l/min'
      , 'l/s'
      , 'lb'
      , 'lbf'
      , 'lx'
      , 'm'
      , 'm/h'
      , 'm/s'
      , 'm/s2'
      , 'm2'
      , 'm3'
      , 'm3/h'
      , 'm3/min'
      , 'm3/s'
      , 'mA'
      , 'mC'
      , 'mcg'
      , 'mg'
      , 'mi'
      , 'mi2'
      , 'min'
      , 'min/km'
      , 'min/mi'
      , 'ml'
      , 'ml/s'
      , 'mm'
      , 'mm2'
      , 'mm3'
      , 'mm3/s'
      , 'month'
      , 'ms'
      , 'msk'
      , 'mt'
      , 'mu'
      , 'nC'
      , 'mHz'
      , 'MHz'
      , 'mV'
      , 'mVA'
      , 'MVA'
      , 'mVAR'
      , 'MVAR'
      , 'mVARh'
      , 'MVARh'
      , 'mW'
      , 'MW'
      , 'mWh'
      , 'MWh'
      , 'nMi'
      , 'ns'
      , 'oz'
      , 'pC'
      , 'pnt'
      , 'pnt/h'
      , 'pnt/min'
      , 'pnt/s'
      , 'ppb'
      , 'ppm'
      , 'ppq'
      , 'ppt'
      , 'psi'
      , 'qt'
      , 'qt/s'
      , 'rad'
      , 'rad/s'
      , 'rpm'
      , 's'
      , 's/m'
      , 's/ft'
      , 't'
      , 'Tbs'
      , 'Tbs/s'
      , 'THz'
      , 'torr'
      , 'tsk'
      , 'tsp'
      , 'tsp/s'
      , 'us-ft'
      , 'V'
      , 'VA'
      , 'VAR'
      , 'VARh'
      , 'W'
      , 'week'
      , 'Wh'
      , 'yd'
      , 'yd2'
      , 'yd3'
      , 'yd3/h'
      , 'yd3/min'
      , 'yd3/s'
      , 'year'
      , 'μC'
      , 'us-in'
      , 'in-us'
  ];

  try {
    assert.deepEqual(actual.sort(), expected.sort());
  }
  catch (e) {
    // This gets too long, and gets truncated
    process.stderr.write(e + '\n');
    throw e;
  }
};

module.exports = tests;


lib/definitions/length.js
var metric,
imperial;

metric = {
  mm: {
    name: {
      singular: 'Millimeter',
      plural: 'Millimeters'
    },
    to_anchor: 1/1000
  },
  cm: {
    name: {
      singular: 'Centimeter',
      plural: 'Centimeters'
    },
    to_anchor: 1/100
  },
  m: {
    name: {
      singular: 'Meter',
      plural: 'Meters'
    },
    to_anchor: 1
  },
  km: {
    name: {
      singular: 'Kilometer',
      plural: 'Kilometers'
    },
    to_anchor: 1000
  }
};

imperial = {
  'in': {
    name: {
      singular: 'Inch',
      plural: 'Inches'
    },
    to_anchor: 1/12
  },
  yd: {
    name: {
      singular: 'Yard',
      plural: 'Yards'
    },
    to_anchor: 3
  },
  'ft-us': {
    name: {
      singular: 'US Survey Foot',
      plural: 'US Survey Feet'
    },
    to_anchor: 1.000002
  },
  'us-ft': {
    name: {
      singular: 'US Survey Foot',
      plural: 'US Survey Feet'
    },
    to_anchor: 1.000002
  },
  ft: {
    name: {
      singular: 'Foot',
      plural: 'Feet'
    },
    to_anchor: 1
  },
  fathom: {
    name: {
      singular: 'Fathom',
      plural: 'Fathoms',
    },
    to_anchor: 6
  },
  mi: {
    name: {
      singular: 'Mile',
      plural: 'Miles'
    },
    to_anchor: 5280
  },
  nMi: {
    name: {
      singular: 'Nautical Mile',
      plural: 'Nautical Miles'
    },
    to_anchor: 6076.12
  },
  'us-in': {
    name: {
      singular: 'US Survey Inch',
      plural: 'US Survey Inches'
    },
    to_anchor: 0.08333316667
  },
  'in-us': {
    name: {
      singular: 'US Survey Inch',
      plural: 'US Survey Inches'
    },
    to_anchor: 0.08333316667
  }
};

module.exports = {
  metric: metric,
  imperial: imperial,
  _anchors: {
    metric: {
      unit: 'm',
      ratio: 3.280839895013123
    },
    imperial: {
      unit: 'ft',
      ratio: 1/3.280839895013123
    }
  }
};
